src/Microsoft.ML.KMeansClustering/KMeansCatalog.cs

```diff
@@ -29,7 +29,7 @@ public static class KMeansClusteringExtensions
         /// </example>
         public static KMeansPlusPlusTrainer KMeans(this ClusteringCatalog.ClusteringTrainers catalog,
            string featureColumn = DefaultColumnNames.Features,
```

Comment:
>featureColumn [](start = 18, length = 13)

So we're calling this `exampleWeightColumnName`, but calling the above `featureColumn`. I'm not terribly insistent one way or the other, but we should at least be consistent I think.

In the original issue you proposed calling this `* exampleWeightColumn`, which seems to me to be consistent with the existing behavior *here*. Yet elsewhere I see that we are calling things `labelColumn` and `labelColumnName` here and there.

Is this on someone's radar? If we are not going to fix it (we should!) a second best alternative would be to be at least locally consistent. Not sure what @rogancarr and @eerhardt and @sfilipi think?

Reply:
Issue #2177  proposes we use suffix *ColumnName consistently  

In this PR I am making that change for  weight, groupid  parameters.  In a follow up PR I will do it for label and features.  

So once  *both* PRs go in, we will mark #2177 as complete, and the following convention will be used
- labelColumnName
- featureColumnName
- exampleWeightColumnName
- rowGroupColumnName

This is certainly on my radar, as i drive towards consistency in the APIs . I have also updated the description of the issue so it has reference to issue #2177 .  Hope that sounds good ?

Reply:
Very good, thank you @abgoswam ! Sorry missed that discussion.

Reply:
@tomfinley. I have consolidated all the changes in this PR.  So  this PR  addresses   {features, labels, groupid, weight} column names in a consistent manner.

---
In reply to: [258779314](https://github.com/dotnet/machinelearning/pull/2665#discussion_r258779314) [](ancestors = 258779314)

Reply:
@tomfinley. I have consolidated all the changes in this PR.  So  this PR  addresses   {features, labels, groupid, weight} column names in a consistent manner.

---
In reply to: [258776812](https://github.com/dotnet/machinelearning/pull/2665#discussion_r258776812) [](ancestors = 258776812)